src/app/project/ontology/property-form/property-form.component.ts

```diff
@@ -375,4 +403,45 @@ export class PropertyFormComponent implements OnInit {
         }
     }
 
+    setCardinality(prop: ResourcePropertyDefinitionWithAllLanguages) {
+
+        const onto = new UpdateOntology<UpdateResourceClassCardinality>();
+
+        onto.lastModificationDate = this.lastModificationDate;
+
+        onto.id = this.ontology.id;
+
+        const addCard = new UpdateResourceClassCardinality();
+
+        addCard.id = this.resClassIri;
+
+        addCard.cardinalities = [];
+
+        const propCard: IHasProperty = {
+            propertyIndex: prop.id,
+            cardinality: this._ontologyService.translateCardinality(this.propertyForm.value.multiple, this.propertyForm.value.required),
+            guiOrder: this.guiOrder  // add new property to the end of current list of properties
+        };
+
+        addCard.cardinalities.push(propCard);
+
+        onto.entity = addCard;
+
+        this._dspApiConnection.v2.onto.addCardinalityToResourceClass(onto).subscribe(
+            (res: ResourceClassDefinitionWithAllLanguages) => {
+
+                this.lastModificationDate = res.lastModificationDate;
+                // close the dialog box
+                this.loading = false;
+                this.closeDialog.emit();
+            },
+            (error: ApiResponseError) => {
+                this._errorHandler.showMessage(error);
+            }
+        );
+
+        this.loading = false;
+        this.closeDialog.emit();
```

Comment:
`this.closeDialog.emit();` should run only once namely only the one on line 436.

Reply:
fixed in b56a43a